Chap_API_Storage.tex

```diff
@@ -0,0 +1,90 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Chapter: Storage support
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\chapter{Storage Support Definitions}
+\label{chap:api_storage}
+
+Distributed and parallel computing systems are increasingly embracing storage hierarchies to meet the diverse data management needs of applications and other systems software in a cost-effective manner.
+These hierarchies provide access to a number of distinct storage layers, with each potentially composed of different storage hardware (e.g., HDD, SDD, tape, NVM), deployed at different locations (e.g., on-node, on-switch, on-site, WAN), and designed using different storage paradigms (e.g., file-based, object-based).
+Each of these systems offers unique performance and usage characteristics that storage system users should carefully consider to ensure the most efficient use of storage resources.
+
+PMIx enables users to better understand storage hierarchies by defining attributes that formalize storage system characteristics, state, and other parameters.
+These attributes can be queried by applications, I/O libraries and middleware, and workflow systems to discover available storage resources and to inform on which resources are most suitable for different I/O workload requirements.
+
+
+%%%%%%%%%%%
+\section{Storage-related attributes}
+\label{api:struct:attributes:pstrg}
+
+\versionMarker{4.0}
+
+The following attributes may be returned in response to queries (e.g., \refapi{PMIx_Get} or \refapi{PMIx_Query_info}) made by processes or tools.
+
+%
+\declareAttributeNEW{PMIX_STORAGE_ID}{"pmix.strg.id"}{char*}{
+An identifier for the storage system (e.g., lustre-fs1, daos-oss1, home-fs)
+}
+
+%
+\declareAttributeNEW{PMIX_STORAGE_PATH}{"pmix.strg.path"}{char*}{
+Mount point path for the storage system (valid only for file-based storage systems)
+}
+
+%
+\declareAttributeNEW{PMIX_STORAGE_TYPE}{"pmix.strg.type"}{char*}{
```

Comment:
This can be a qualifier, but is also something that can be returned as a value, yes? I can imagine someone asking "what is the storage type on this mount point". As written, it indicates that this can only be used as a qualifier.

Reply:
Good call, I'll correct it so it's clear it's not just a qualifier, but a general attribute. 

Related to this and the 2 previous conversations, but is there a convention that should be followed to specify whether things are attributes, qualifiers, or both? Just something in the text description to help make it obvious? I was thinking it could be useful to understand what types of qualifiers are valid for different types of attributes, but don't see things organized like that elsewhere in the standard. That would help make it clear whether an attribute is something you can query, something you use to qualify, or both.

Reply:
No real convention - I just indicate it in the text if there is something unusual, e.g., that it can only be used as a qualifier. If it can be used as both a key to a query and as a qualifier, then I don't bother to say anything. I only raised the point because you had specifically called out this one attribute to be only a qualifier and wasn't sure why.